api.ts

```diff
@@ -547,11 +546,11 @@ export function makeFormula<ParamDefsT extends ParamDefs>(
       break;
     }
     case ValueType.Number: {
-      const {resultType: unused, codaType, ...rest} = definition;
```

Comment:
This seems like a breaking change to start ignoring `codaType` at the top level of the formula definition. That was introduced as a shorthand so makers wouldn't have to confront the `schema` field in most cases. It's being used in a lot of pack defs so we can't just ignore. I think the answer is probably to use the schema if it's present, otherwise use `codaType` in a synthetic schema like it's currently doing.

Reply:
Yeah that makes sense. I think the side effect of keeping that synthetic schema is that
```
{type: ValueType.Number, codaType}
```
is not valid for scale schemas (maximum and icon are required, and our coda parsing code has no fallback for creating the corresponding value format here. So we'd need the following extra steps:
- make `icon` and `maximum` optional properties
- in `coda` we can add a code change to default fallback to `icon: star` and `maximum: 5` or something (as long as the doc doesn't break if the result is greater than the maximum value). How does that sound?

Or alternatively we can add icon and maximum into the synthetic schema when the provided codaType is scale.

Reply:
Ah interesting, the scale properties are required by the similar ones for slider are optional. Yeah, we should do what you suggest, make the scale properties optional and add defaults on the `coda` side. 

There's potentially a way to use existing defaults on the `coda` side, because if you just add a new slider column it applies defaults for you, we should try to trigger that defaulting or reuse the same default values.